src/System.Private.CoreLib/src/System/Threading/Tasks/AsyncCausalityTracer.cs

```diff
@@ -33,8 +33,7 @@ internal static bool LoggingOn
         //Indicates this information comes from the BCL Library
         private const WFD.CausalitySource s_CausalitySource = WFD.CausalitySource.Library;
 
-        //Lazy initialize the actual factory
-        private static WFD.IAsyncCausalityTracerStatics s_TracerFactory;
+        private static WFD.IAsyncCausalityTracerStatics s_TracerFactory = null!;
```

Comment:
Why "null!"?

Reply:
This is one of those cases where the compiler can't see that field will only ever be used when it's non-null.  It's initialized in the cctor, but there's a fast-path exit as well as an error path that bails if something goes wrong, in which case it'll never be initialized.  This is just a way of silencing the compiler's warnings.